### PR TITLE
Update Smart.simba

### PR DESCRIPTION
**Pull request by BMWxi**
Added a the option to not show the SMART console.
Script usage: Put `SmartShowConsole := True;` before `Reflect.setup;`
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23issuecomment-74380653%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23discussion_r24715123%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23discussion_r24715126%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23discussion_r24715171%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23discussion_r24715172%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23issuecomment-74380718%22%2C%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23issuecomment-77716068%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23issuecomment-74380653%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22done%22%2C%20%22created_at%22%3A%20%222015-02-14T16%3A08%3A13Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7343144%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/BMWxi%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%22%2C%20%22created_at%22%3A%20%222015-02-14T16%3A10%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7343144%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/BMWxi%22%7D%7D%2C%20%7B%22body%22%3A%20%22This%20is%20needed%21%22%2C%20%22created_at%22%3A%20%222015-03-07T23%3A24%3A14Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/8165365%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/SRLTheMayor%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20732680a86dce8615749d9e66224a68b09234aa07%20lib/internal/Smart.simba%2021%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23discussion_r24715126%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22No%20need%20for%20%27%28%29%27%22%2C%20%22created_at%22%3A%20%222015-02-14T16%3A01%3A18Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-02-14T16%3A09%3A48Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7343144%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/BMWxi%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/internal/Smart.simba%3AL96-107%22%7D%2C%20%22Pull%20732680a86dce8615749d9e66224a68b09234aa07%20lib/internal/Smart.simba%2012%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/Elfyyy/OSR-Reflection/pull/40%23discussion_r24715123%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Make%20%27String%27%20-%3E%20%27string%27%22%2C%20%22created_at%22%3A%20%222015-02-14T16%3A00%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/4069146%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/Elfyyy%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-02-14T16%3A09%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7343144%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/BMWxi%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20lib/internal/Smart.simba%3AL86-93%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/40#issuecomment-74380653'>General Comment</a></b>
- <a href='https://github.com/BMWxi'></a> done
- <a href='https://github.com/BMWxi'></a> :+1:
- <a href='https://github.com/SRLTheMayor'></a> This is needed!
- [x] <a href='#crh-comment-Pull 732680a86dce8615749d9e66224a68b09234aa07 lib/internal/Smart.simba 12'></a> &nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/40#discussion_r24715123'>File: lib/internal/Smart.simba:L86-93</a></b>
- <a href='https://github.com/Elfyyy'></a> Make 'String' -> 'string'
- [x] <a href='#crh-comment-Pull 732680a86dce8615749d9e66224a68b09234aa07 lib/internal/Smart.simba 21'></a> &nbsp;<b><a href='https://github.com/Elfyyy/OSR-Reflection/pull/40#discussion_r24715126'>File: lib/internal/Smart.simba:L96-107</a></b>
- <a href='https://github.com/Elfyyy'></a> No need for '()'

<a href='https://www.codereviewhub.com/Elfyyy/OSR-Reflection/pull/40?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/Elfyyy/OSR-Reflection/pull/40?approve=1'></a>&nbsp;<a href='https://github.com/Elfyyy/OSR-Reflection/pull/40'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by Elfyyy** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/Elfyyy/OSR-Reflection/pull/40'>Refresh</a>.<a href='#crh-update-none'></a>

File: lib/internal/Smart.simba [hunk 1]
```diff
@@ -85,6 +86,7 @@ procedure TReflectionSmart.Create(
   Width, Height: Integer; UserAgent: string); overload;
 var
   World: Variant;
+  JavaString: String;
```
Comment (KyleHunter) [inline lib/internal/Smart.simba]: Make 'String' -> 'string'

File: lib/internal/Smart.simba [hunk 2]
```diff
@@ -94,7 +96,11 @@ begin
     Exit;
   World := Self.GetWorld;
   Reflect.Logger.Status('Using world %s.', [World]);
-  Self.Target := SmartSpawnClient('java.exe', StringReplace(PluginPath, '\',
+  if (SmartShowConsole) then
```
Comment (KyleHunter) [inline lib/internal/Smart.simba]: No need for '()'

**Comment by BMWxi** [comment]
done

File: lib/internal/Smart.simba (hunk 2 above)
Comment (BMWxi) [inline lib/internal/Smart.simba]: :+1:<a href='#crh-update-none'></a>

File: lib/internal/Smart.simba (hunk 1 above)
Comment (BMWxi) [inline lib/internal/Smart.simba]: :+1:<a href='#crh-update-complete'></a>

**Comment by BMWxi** [comment]
:+1:

**Comment by TheTS** [comment]
This is needed!

**Comment by KyleHunter** [comment]
Make a new one, I can't merge this current rev

**Pull request closed without merging**